### ISSUE TITLE
MLMD import before TF causes graph issue

### ISSUE THREAD

**Issue by eggie5** [google/ml-metadata#101]
See the below self-contained example, where importing MLMD before the TF import causes some type of graph issue:

```
2021-01-25 13:56:36.824683: E tensorflow/core/grappler/optimizers/meta_optimizer.cc:533] model_pruner failed: Invalid argument: MutableGraphView::MutableGraphView error: graph has multiple nodes with the name '/job:localhost/replica:0/task:0/device:CPU:0'.
2021-01-25 13:56:37.393309: W tensorflow/core/grappler/utils.cc:84] Duplicated node in the graph: /job:localhost/replica:0/task:0/device:CPU:0
2021-01-25 13:56:37.393357: F ./tensorflow/core/grappler/graph_view.h:332] Check failed: st.ok() Non unique node name detected: /job:localhost/replica:0/task:0/device:CPU:0
```

* OSX 10.14
* TF: 1.15.2
* py37
* protobuf 3.14.0 (tried all other versions)

```python
# Causes Bug 😭
# from ml_metadata import metadata_store
# from ml_metadata.proto import metadata_store_pb2

import tensorflow as tf

# OK 👍
from ml_metadata import metadata_store
from ml_metadata.proto import metadata_store_pb2

import pandas as pd



CSV_COLUMN_NAMES = ['SepalLength', 'SepalWidth', 'PetalLength', 'PetalWidth', 'Species']
SPECIES = ['Setosa', 'Versicolor', 'Virginica']

train_path = tf.keras.utils.get_file(
    "iris_training.csv", "https://storage.googleapis.com/download.tensorflow.org/data/iris_training.csv")
test_path = tf.keras.utils.get_file(
    "iris_test.csv", "https://storage.googleapis.com/download.tensorflow.org/data/iris_test.csv")

train = pd.read_csv(train_path, names=CSV_COLUMN_NAMES, header=0)
test = pd.read_csv(test_path, names=CSV_COLUMN_NAMES, header=0)

train_y = train.pop('Species')
test_y = test.pop('Species')

# The label column has now been removed from the features.
train.head()


def input_evaluation_set():
    features = {'SepalLength': np.array([6.4, 5.0]),
                'SepalWidth':  np.array([2.8, 2.3]),
                'PetalLength': np.array([5.6, 3.3]),
                'PetalWidth':  np.array([2.2, 1.0])}
    labels = np.array([2, 1])
    return features, labels
    
def input_fn(features, labels, training=True, batch_size=256):
    """An input function for training or evaluating"""
    # Convert the inputs to a Dataset.
    dataset = tf.data.Dataset.from_tensor_slices((dict(features), labels))

    # Shuffle and repeat if you are in training mode.
    if training:
        dataset = dataset.shuffle(1000).repeat()

    return dataset.batch(batch_size)
    
    
# Feature columns describe how to use the input.
my_feature_columns = []
for key in train.keys():
    my_feature_columns.append(tf.feature_column.numeric_column(key=key))
    
# Build a DNN with 2 hidden layers with 30 and 10 hidden nodes each.
classifier = tf.estimator.DNNClassifier(
    feature_columns=my_feature_columns,
    # Two hidden layers of 30 and 10 nodes respectively.
    hidden_units=[30, 10],
    # The model must choose between 3 classes.
    n_classes=3)
    
# Train the Model.
classifier.train(
    input_fn=lambda: input_fn(train, train_y, training=True),
    steps=5000)
    
eval_result = classifier.evaluate(
    input_fn=lambda: input_fn(test, test_y, training=False))

print('\nTest set accuracy: {accuracy:0.3f}\n'.format(**eval_result))
```

**Comment by hughmiao** [comment]
hi Alex, thanks for reporting this. This is a known issue. Currently the mlmd C++ lib depends on part of the tf 1.15 core lib, we are deprecating mlmd C++ TF dependency to resolve this. Let's keep this bug open until fixed.

**Comment by hughmiao** [comment]
this is fixed in 0.29 release. please feel free to reopen.

**Comment by duduscript** [comment]
Hi @hughmiao , will ml_metadata support tensorflow version <= 1.15.2 ? I notice there is no dependency on tensorflow version .

**Comment by hughmiao** [comment]
@duduscript , we deprecated the tf py dependency. You can use the released pip wheel >=0.29 with any tf version.

**Comment by duduscript** [comment]
Thank you @hughmiao , I test it in tf1.12 and it works .
